### PR TITLE
Add Support for Einstein Probe Alerts and HEALPix Processing Script

### PR DESCRIPTION
**Pull request by nkosogor**
@caseyjlaw Can you review these changes:

- Added support for Einstein Probe alerts in `gcn_kafka_receiver.py`+ switched to IPv4 for Kafka client:
https://github.com/ovrocaltech/ovro-alert/blob/e72907a4904c69f88cf487e6072317b103b257b3/gcn-alert/gcn_kafka_receiver.py#L36-L46
- Implemented the `analyze_map.py` script for processing HEALPix sky maps related to LIGO events. The script:
  - Accepts the time of observation and a FITS file.
  - Outputs three maps: the event probability map, the observable sky map for LWA at the time of observation, and a probability map masked with the observable sky.
  - Prints the total 90% credible region area and the 90% credible region area observable by LWA

### CODE REVIEW

File: gcn-alert/gcn_kafka_receiver.py [hunk 1]
```diff
@@ -56,12 +67,15 @@ def post_to_slack(channel, message):
...
+            rate_duration = alert.get("rate_duration", None)
+            event_time_str = alert.get("trigger_time", None)
+            event_time = datetime.strptime(event_time_str, '%Y-%m-%dT%H:%M:%S.%fZ') if event_time_str else None
             current_time = datetime.utcnow()
 
             if (
-                rate_duration < 2 
+                rate_duration is not None 
+                and event_time is not None
+                and rate_duration < 2
```
Comment (caseyjlaw) [inline gcn-alert/gcn_kafka_receiver.py:78]: if `rate_duration` is `None`, then I think this will produce a `TypeError`. Need to test for `None` before testing if value is less than 2.

File: gcn-alert/gcn_kafka_receiver.py [hunk 2]
```diff
@@ -86,6 +100,35 @@ def post_to_slack(channel, message):
                 )
                 if send_to_slack:
                     post_to_slack(slack_channel, message)
+            elif (
+                "ra" in alert
+                and "dec" in alert
+                and "ra_dec_error" in alert
+                and "net_count_rate" in alert
+                and "image_snr" in alert
+                and event_time is not None
```
Comment (caseyjlaw) [inline gcn-alert/gcn_kafka_receiver.py:109]: As above, I think you need to test for `None` before you can do a comparison.

File: ligo-alert/analyze_map.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    i = np.flipud(np.argsort(hpx))
+    sorted_credible_levels = np.cumsum(hpx[i])
+    credible_levels = np.empty_like(sorted_credible_levels)
+    credible_levels[i] = sorted_credible_levels
+
+    credible_area = np.sum(credible_levels <= 0.9) * hp.nside2pixarea(nside, degrees=True)
+    print(f"90% credible region area: {credible_area:.2f} square degrees")
+
+    # Define the observatory's ECEF coordinates
+    x = -2409261.7339418 * u.m
```
Comment (caseyjlaw) [inline ligo-alert/analyze_map.py:28]: Not critical, but we have software to manage the OVRO-LWA antenna coordinates.

```
import lwa_antpos
lon, lat = lwa_antpos.lwa_df.loc['LWA-000'][['longitude', 'latitude']].to_list()
EarthLocation.from_geodetic(lon, lat)
```

**Review by caseyjlaw — APPROVED** [review]
I added small comments, but it looks good.
I am not familiar with how the certificates work, but I think all you need to know that it runs?

**Review by caseyjlaw — APPROVED** [review]

**Comment by caseyjlaw** [comment]
I reinstalled to the py36 and py39 environments. Then I restarted the chime, ligo, and gcn services.
I think the gcn service is being run from your fork, though. I think you'll need to pull the code yourself and reinstall/restart the gcn service for that change to hold.
